openlibrary/macros/LoanStatus.html

```diff
@@ -1,22 +1,27 @@
-$def with (page, ebook_text, user, current_and_available_loans, contributor, line_break = '', editions_page=False, isbn=None, oclc_numbers=None)
-
-$# TODO construct list of available formats from available_loans
+$def with (page, user, editions_page=False)
 
 $ work = page.works and page.works[0]
-
-$if current_and_available_loans:
-  $ current_loans = current_and_available_loans[0]
-$else:
-  $ current_loans = []
-
+$ meta_fields = page.get_ia_meta_fields() or {}
+$ contributor = meta_fields.get('contributor')
+$ user_loan = None
+$ waiting_loan = ctx.user and ctx.user.get_waiting_loan_for(page)
+$ my_turn_to_borrow = waiting_loan and waiting_loan['status'] == 'available' and waiting_loan['position'] == 1
 $ realtime_availability = page.get_realtime_availability()
 $ availability = realtime_availability.get('status', 'error')
 $ wlsize = realtime_availability.get('num_waitlist', 0)
-$ waiting_loan = ctx.user and ctx.user.get_waiting_loan_for(page)
-$ my_turn_to_borrow = waiting_loan and waiting_loan['status'] == 'available' and waiting_loan['position'] == 1
+$ current_and_available_loans = []
 
+$if page.get('ocaid'):
+  $ current_and_available_loans = page.get_current_and_available_loans()
+  $ user_loan = None
+  $if ctx.user:
+    $for current_loan in current_and_available_loans[0]:
+      $if current_loan['user'] == ctx.user.key:
+        $ user_loan = current_loan
+        $ break
```

Comment:
SUGGESTION: No space needed, I think, between `$` and `break`.

Reply:
I think this may be required, otherwise it treats `break` as a variable, I think this is a weird quirk (http://webpy.org/templetor)